### DISCUSSION TITLE
Open Telemetry and Prisma query events, missing context?

### DISCUSSION THREAD

**Discussion by dariusj18** [prisma/prisma#8926] [Q&A]
I am trying to log the Prisma query events inside a OpenTelemetry trace, however when I try and get the current span or context it seems that there is nothing there.

```
prisma.$on<any>('query', (e: Prisma.QueryEvent) => {
	const span = otelApi.trace.getSpan(otelApi.context.active());

	span.addEvent('query', {
		'query.sql': e.query,
		'query.duration': e.duration,
	});
});
```

but everything is fine when I try and add to the trace in a middleware defined just above this.

Does this mean it's just not data I can access inside an event?

**Comment by MikePrynAvenga** [comment]
@dariusj18, hi! I am having the same issue. Have you managed to work around this issue?

**Reply by dariusj18** [reply]
No, I have not gone back to telemetry since

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
